eventsources/sources/awssqs/start.go

```diff
@@ -112,6 +111,10 @@ func (el *EventListener) StartListening(ctx context.Context, dispatch func([]byt
 		messages, err := fetchMessages(ctx, sqsClient, *queueURL.QueueUrl, 10, sqsEventSource.WaitTimeSeconds)
 		if err != nil {
 			log.Errorw("failed to get messages from SQS", zap.Error(err))
+			_, newSqsClient, err := el.refreshSessionAndSqsClientIfCredentialsExpired(err, log)
```

Comment:
Can we check the error here, if the error is `expired`, reuse the logic above to get a new client? You can wrap the code above to a function.

Reply:
I've made the change to check the error here